src/graphql/node/status.rs

```diff
@@ -36,195 +36,123 @@ impl NodeStatusQuery {
+
+        let (resource_usage, ping) =
+            fetch_resource_usage_and_ping(agent_manager, hostname, is_manager).await;
+
+        let (features, learner, models) =
+            determine_agent_statuses(&online_agents_by_host_id, hostname, &node.agents);
+
         connection.edges.push(Edge::new(
-            crate::graphql::encode_cursor(&ev.unique_key()),
+            crate::graphql::encode_cursor(&node.unique_key()),
             NodeStatus::new(
-                ev.id,
-                ev.name,
-                cpu_usage,
-                total_memory,
-                used_memory,
-                total_disk_space,
-                used_disk_space,
+                node.id,
+                node.name,
+                resource_usage.as_ref().map(|x| x.cpu_usage),
+                resource_usage.as_ref().map(|x| x.total_memory),
+                resource_usage.as_ref().map(|x| x.used_memory),
+                resource_usage.as_ref().map(|x| x.total_disk_space),
+                resource_usage.as_ref().map(|x| x.used_disk_space),
                 ping,
-                review,
-                piglet,
-                piglet_config,
-                reconverge,
-                hog,
-                hog_config,
+                is_manager,
+                features,
+                learner,
+                models,
             ),
         ));
     }
     Ok(connection)
 }
 
+// Returns the resource usage and ping time of the given hostname.
+async fn fetch_resource_usage_and_ping(
+    agent_manager: &BoxedAgentManager,
+    hostname: &str,
+    is_manager: bool,
+) -> (Option<ResourceUsage>, Option<Duration>) {
+    if is_manager {
+        // Since this code is executed on the Manager server itself, we retrieve the resource
+        // usage directly without making a remote call. The ping value is set to 0 without
+        // performing an actual ping, because ping on the same machine should result in negligible
+        // round-trip time (RTT).
+        (
+            Some(roxy::resource_usage().await),
+            Some(Duration::from_secs(0)),
+        )
+    } else {
+        (
+            agent_manager.get_resource_usage(hostname).await.ok(),
+            agent_manager.ping(hostname).await.ok(),
+        )
+    }
+}
+
+// Returns the statuses of the Features, Learner, and Models agents for the given hostname.
+fn determine_agent_statuses(
+    online_agents_by_host_id: &HashMap<String, Vec<(String, String)>>,
+    hostname: &str,
+    agents: &[review_database::Agent],
+) -> (AgentStatus, AgentStatus, AgentStatus) {
+    // TODO: The current logic determines an agent's status solely based on its presence in the
+    // `online_agents_by_host_id` map. An agent is marked as `Enabled` if found in the map, and
+    // `Disabled` if not. In the future, we may consider implementing more advanced logic that
+    // compares the agent's status in the DB with its online status.
```

Comment:
Could you post an issue regarding this instead of leaving a comment? If a
comment is necessary to alert others, we can simply reference the issue number. BTW,
I don't quite understand what "its online status" means, but I think agents should update
their status to the manager whenever they change, which hasn't been implemented yet.